### ISSUE TITLE
commandline randomness found the bug

### ISSUE THREAD

**Issue by gcmartijn** [aseprite/aseprite#1232]
The commandline has a random order when or when not some options are working.

For example this works (json-array output)
/Applications/Aseprite.app/Contents/MacOS/aseprite \
--batch \
--format json-array ./Aseprite/lopen/looprechts.ase ./Aseprite/lopen/loopboven.$
--filename-format '{title}_{frame}' \
--trim \
--shape-padding \
--ignore-empty \
--sheet-pack \
--sheet ./MX2/assets/atlas-player.png --data ./MX2/assets/atlas-player.json \

But this don't work (default json-hash output)
/Applications/Aseprite.app/Contents/MacOS/aseprite \
--batch ./Aseprite/lopen/looprechts.ase ./Aseprite/lopen/loopboven.$
--filename-format '{title}_{frame}' \
--trim \
--format json-array
--shape-padding \
--ignore-empty \
--sheet-pack \
--sheet ./MX2/assets/atlas-player.png --data ./MX2/assets/atlas-player.json \

This works to get all the layers in the json file
/Applications/Aseprite.app/Contents/MacOS/aseprite \
--batch \
--split-layers \
--format json-array ./Aseprite/scene1.ase \
--filename-format '{layer}' \
--shape-padding \
--ignore-empty \
--sheet-pack \
--trim \
--sheet ./MX2/assets/atlas-world1.png --data ./MX2/assets/atlas-world1.json \

This don't give any layer output
/Applications/Aseprite.app/Contents/MacOS/aseprite \
--batch ./Aseprite/scene1.ase \
--split-layers \
--format json-array 
--filename-format '{layer}' \
--shape-padding \
--ignore-empty \
--sheet-pack \
--trim \
--sheet ./MX2/assets/atlas-world1.png --data ./MX2/assets/atlas-world1.json \

I don't know how many commands don't work at what order they are.
In other words, or update the documentation in what order you need to use, or its a bug.

:)

### Aseprite and System version
osx 1.1.7 and others

**Comment by dacap** [comment]
Hi @gcmartijn 

On the case of `--split-layers`, the option must be used before specifying the `.ase` file. I'm not sure if the option is well documented, [here](http://www.aseprite.org/docs/cli/#options) or [here](http://www.aseprite.org/docs/cli/#split-layers), the documentation says:

> Import each layer of the next given sprite as a separated image in the sheet

And

> Splits the next document layers

--

As a side note, I saw that you've used the `--shape-padding` argument without a `<value>`, that option must be used with an integer value (e.g. `--shape-padding 2` to add a padding of 2). It needs [better documentation](http://www.aseprite.org/docs/cli/#shape-padding) (in the [Options section](http://www.aseprite.org/docs/cli/#options) it's more clear the extra `<value>` parameter).

--

About the `--format json-array`, I'm not sure what's happening in that one, I'll test this tomorrow. Anyway you can give a try to the beta branch on Steam, it has a `--preview` parameter to print/show/check what Aseprite will do on the console.

**Comment by gcmartijn** [comment]
@dacap 
Oke, I think that was the problem.
Don't have steam but for now I can work with it ;)
